sdk/appconfiguration/app-configuration/src/internal/helpers.ts

```diff
@@ -232,42 +231,40 @@ export function serializeAsConfigurationSettingParam(
 /**
  * @internal
  */
-export function transformKeyValueResponseWithStatusCode<
-  T extends KeyValue & HttpResponseField<any>
->(kvp: T): ConfigurationSetting & { eTag?: string } & HttpResponseField<any> & HttpResponseFields {
-  return normalizeResponse(kvp, <
-    ConfigurationSetting & HttpResponseField<any> & HttpResponseFields
-  >{
+export function transformKeyValueResponseWithStatusCode<T extends KeyValue>(
+  kvp: T,
+  status: number | undefined
+): ConfigurationSetting & { eTag?: string } & HttpResponseFields {
+  const response = {
     ...transformKeyValue(kvp),
-    statusCode: kvp._response.status,
-  });
+    statusCode: isDefined(status) ? status : -1,
+  };
+
+  if (hasUnderscoreResponse(kvp)) {
+    Object.defineProperty(response, "_response", {
+      enumerable: false,
```

Comment:
not technically required (all flags default to false), but I'm fine with making it explicit

Reply:
TIL